docs/2-developing/0-connectors/3-connector-configuration.mdx

```diff
@@ -0,0 +1,299 @@
+---
+title: "Connector Configuration"
+---
+
+A source or destination connector's configuration is provided through a struct,
+with typed parameter values, which makes it simple to work with throughout the
+connector code.
+
+Here's an example:
+
+```go
+type SourceConfig struct {
+    sdk.UnimplementedSourceConfig
```

Comment:
Most, if not all connectors should use `sdk.DefaultSourceMiddleware`, so I'd steer our users towards that. Using `sdk.UnimplementedSourceConfig` should only be done in case you know what you're doing and intentionally don't want middleware.

So I'd turn it around and change all examples to use `sdk.DefaultSourceMiddleware`, then call it out in the section about middleware that you can _turn it off_ by embedding `sdk.UnimplementedSourceConfig`, although the regular user should probably not do that unless they have a good reason.

Reply:
Good thinking. Yeah, I initially wanted something minimal, but it doesn't really help here.

Reply:
Done!